core/core-generic-ops.el

```diff
@@ -0,0 +1,144 @@
+(require 'evil)
+
+;; Evaling is for modes who can send text to a repl and receive a response
+
+(defvar spacemacs--generic-eval-region-fallback-to-send t
+  "When truthy, fallback to using spacemacs--generic-send-region-alist if no eval func exists for mode")
+
+(defvar spacemacs--generic-eval-region-alist '()
+  "Alist mapping mode to a 2-arity function which evals from beg to end for the mode")
+
+(defun spacemacs//get-eval-region-for-current-mode ()
+  (interactive)
```

Comment:
Interactive function should use `/` instead of `//` in their names. `//` is for implementation details functions.
Also we try to document all interactive function with a docstring ;-)